include/benchmark/benchmark.h

```diff
@@ -982,11 +980,9 @@ class Benchmark {
   //
   // The callback will be passed the number of threads for this benchmark run.
   //
-  // The callback must not be self-deleting.  The Benchmark
-  // object takes ownership of the callback std::function object.
-  Benchmark* SetUp(std::function<void(benchmark::State&)>);
-  Benchmark* TearDown(std::function<void(benchmark::State&)>);
-#endif
+  // The callback must not be NULL or self-deleting.
+  Benchmark* Setup(void (*setup)(benchmark::State&));
```

Comment:
should the parameter to the setup/teardown methods be `const`?  i can't imagine we want to change the state during setup or teardown.  it should be safe to, but there's a possibility of breaking some implicit invariant if `State` becomes more complicated.

Reply:
Good question - hadn't thought of it. I just made it non-const to be consistent (`state` has been tossed around as non-const ref everywhere) 
Practically speaking, the state object passed to these callbacks is thrown away after they're invoked so changes on it has no effect.

How strongly do you feel about changing this to const?

Reply:
if it's thrown away, even more strongly. not because it could cause a bug but because it indicates the user _can't_ change it prior to the benchmark being called.

Reply:
Good point! Done! :)